yjit.c

```diff
@@ -304,6 +310,42 @@ rb_get_cme_def_body_optimized_type(rb_callable_method_entry_t* cme)
     return cme->def->body.optimized.type;
 }
 
+rb_method_cfunc_t*
+rb_get_cme_def_body_cfunc(rb_callable_method_entry_t* cme)
+{
+    return UNALIGNED_MEMBER_PTR(cme->def, body.cfunc);
+}
+
+int
+rb_get_mct_argc(rb_method_cfunc_t *mct)
+{
+    return mct->argc;
+}
+
+void*
+rb_get_mct_func(rb_method_cfunc_t *mct)
+{
+    return (void*)mct->func; // this field is defined as type VALUE (*func)(ANYARGS)
+}
+
+int
+rb_get_builtin_argc(struct rb_builtin_function* bi)
+{
+    return bi->argc;
+}
+
+void*
+rb_get_builtin_func_ptr(struct rb_builtin_function* bi)
+{
+    return bi->func_ptr;
+}
```

Comment:
@XrXr maybe we should export some of these structs because it's going to become hard to keep track of what these getters do.

Reply:
This is going to make for unreadable code :/ Not your fault Noah, just saying not exporting the C struct defs is going to make the Rust code hard to read.

Reply:
I agree. This is what I was worried about when we talked earlier about having the little C getters - there aren't as many as I was afraid of, but there are a *lot* and they're pretty hard to keep track of. Exporting the C struct defs should be okay in terms of future changes as long as we regenerate bindings for them frequently, I think.